src/opensles/AudioOutputStreamOpenSLES.cpp

```diff
@@ -129,6 +129,11 @@ Result AudioOutputStreamOpenSLES::open() {
                   AudioFormat::I16 : AudioFormat::Float;
     }
 
+    // OpenSL ES only supports I16 and Float
+    if (mFormat != AudioFormat::I16 && mFormat != AudioFormat::Float) {
+        return Result::ErrorInvalidFormat;
+    }
```

Comment:
The format validation is the same for input and output. In that case, why not put this in AudioStreamOpenSLES::open?

Reply:
Change now in a different spot. At this new spot, I don't think it should be in AudioStreamOpenSLES::open